mlcomp/parallelm/pipeline/component_info.py

```diff
@@ -0,0 +1,319 @@
+from parallelm.pipeline import json_fields
+
+from parallelm.pipeline.component_connection_info import ComponentConnectionInfo
+from parallelm.pipeline.component_argument_info import ComponentArgumentInfo
+from collections import OrderedDict
+
+import json
+import six
+
+
+class ComponentInfoEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, ComponentInfo):
+            ret = OrderedDict()
+            if obj.name:
```

Comment:
If obj is a dict, couldn't you just to the following:
`
for k, v in obj.itmes():
    if v: ret[k] = v
`

Reply:
Obj is not dict, obj is `ComponentInfo`:
used like this `json.dumps(component_info, indent=4, cls=ComponentInfoEncoder)`

Reply:
I am sure there is a nice way to go over all the members and do the big IF using a loop?

Reply:
Maybe... But in our case  Python variable names don't match json keys: `use_mlops` vs `useMLOps`